File: app/src/main/java/com/task/paginationlist/di/modules/DomainModule.java
package com.task.paginationlist.di.modules;

import com.task.paginationlist.domain.interactors.WallpaperListInteractor;
import com.task.paginationlist.repositories.WallpaperRepository;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class DomainModule {

    @Singleton
    @Provides
    public WallpaperListInteractor provideWallpaperInteractor(WallpaperRepository repository) {
        return new WallpaperListInteractor(repository);
    }
}

File: app/src/main/java/com/task/paginationlist/di/modules/DomainModule.kt
package com.task.paginationlist.di.modules

import com.task.paginationlist.domain.interactors.WallpaperListInteractor
import com.task.paginationlist.repositories.WallpaperRepository
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class DomainModule {

    @Singleton
    @Provides
    fun provideWallpaperInteractor(repository: WallpaperRepository): WallpaperListInteractor {
        return WallpaperListInteractor(repository)
    }
}
